### PR TITLE
New tooltips

### PR DESCRIPTION
**Pull request by KutnerUri**
## Proposed Changes

- Use glorified `@tippy.js/react` for tooltips
- New design, Redigma approved
- Replace react-tooltip with new tooltip.
- Add mount point to place all tooltips outside the normal flow (so they won't get affected by scaling, links, etc)

great success

### CODE REVIEW

File: workspace.jsonc [hunk 1]
```diff
@@ -259,6 +260,7 @@
         "subscriptions-transport-ws": "^0.9.16",
         "supi": "0.42.0",
         "terser-webpack-plugin": "^4.1.0",
+        "tippy.js": "^6.2.7",
         "typedoc": "0.18.0",
         "typescript-language-server": "0.4.0",
         "unist-util-remove": "2.0.1",
```
Comment (odedre) [inline workspace.jsonc:266]: remove `react-tooltip` if its not used anymore

**Review by odedre — CHANGES_REQUESTED** [review]

File: scopes/component/ui/component-tooltip/component-tooltip.tsx [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
   status?: ComponentStatus;
-  name: string;
   issuesCount?: number;
 };
 
 // TODO - how do I get the status type without tying this to workspace?
-export function StatusTooltip({ status, name, issuesCount }: any) {
-  if (!status) return null;
+export function StatusTooltip({ status, issuesCount, children }: any) {
+  if (!status) return children;
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/component/ui/component-tooltip/component-tooltip.tsx:13]: No: children are the target for the tooltip (`N`, `M`, etc), and not its content.
Returning them without the tooltip will just render them without the tooltip.

File: scopes/design/ui/tooltip/index.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export { Tooltip } from './tooltip';
+export type { TooltipProps } from './tooltip';
+export { ProvideTooltipInstance, useCtxTooltipInstance } from './shared-instance';
+export type { ProvideTooltipInstanceProps as ProviderProps } from './shared-instance';
+export { SingletonTooltipProvider } from './singleton-instance';
+export type { SingletonTooltipProviderProps } from './singleton-instance';
+export { TooltipMountPoint } from './mount-point';
+
+export { useSingleton, tippy } from '@tippyjs/react';
+export { UseSingletonProps } from '@tippyjs/react';
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/design/ui/tooltip/index.ts:10]: idk. Let's talk about it.

File: scopes/design/ui/tooltip/tooltip.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
+$bg: #2b2b2b;
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/design/ui/tooltip/tooltip.scss:1]: I will update!

[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/design/ui/tooltip/tooltip.scss:1]: Not sure it's the right use case though: it's a certain type background.
Actually, I could apply [dark-theme](https://bit.dev/teambit/base-ui/theme/dark-theme) on it. 🤔 
That would also make any components we may put inside of it (like buttons) dark.

File: scopes/design/ui/tooltip/tooltip.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import React, { ReactElement } from 'react';
+import Tippy, { TippyProps } from '@tippyjs/react';
+import { roundArrow } from 'tippy.js';
+import 'tippy.js/dist/tippy.css';
+import 'tippy.js/dist/svg-arrow.css';
+
+import { getMountPoint } from './mount-point';
+import './tooltip.scss';
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/design/ui/tooltip/tooltip.tsx:8]: The vanilla component is Tippy itself.
It's using html attribute to style theme, so it's global.

File: scopes/design/ui/tooltip/tooltip.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+export function Tooltip(props: TooltipProps) {
+  const ctxInstance = useCtxTooltipInstance();
+
+  const singleton = props.singleton || ctxInstance;
+
+  // children should accept a ref
+  const children = typeof props.children === 'string' ? <span>{props.children}</span> : props.children;
+
+  return (
+    <Tippy arrow={roundArrow} theme={THEME} interactive appendTo={getMountPoint} {...props} singleton={singleton}>
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/design/ui/tooltip/tooltip.tsx:27]: yes, both are overridable!

File: scopes/envs/ui/env-icon/env-icon.tsx [hunk 7]
```diff
@@ -1,10 +1,13 @@
-import React from 'react';
+import React, { forwardRef } from 'react';
 import { ComponentModel } from '@teambit/component';
 
 export type EnvIconProps = { component: ComponentModel } & React.HTMLAttributes<HTMLImageElement>;
 
-export function EnvIcon({ component, ...rest }: EnvIconProps) {
-  if (!component || !component.environment?.icon) return null;
+export const EnvIcon = forwardRef<HTMLImageElement, EnvIconProps>(function EnvIcon(
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/envs/ui/env-icon/env-icon.tsx:6]: so eslint would not say I'm using anonymous components.

[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/envs/ui/env-icon/env-icon.tsx:6]: yes apparently JS/TS is cool with it

**Review by odedre — APPROVED** [review]

**Pull request merged**